apps/condo/domains/common/components/GraphQlSearchInput/GraphQlSearchInput.tsx

```diff
@@ -160,40 +184,38 @@ export const GraphQlSearchInput: React.FC<ISearchInputProps> = (props) => {
                 logEvent({ eventName, eventProperties: componentProperties })
             }
         }
-        handleSearch('')
-    }, [onSelect, props.mode])
-
-    useEffect(() => {
-        handleSearch('')
-    }, [])
-
-    const handleClear = useCallback(() => {
-        setSelected('')
-    }, [])
+        if (autoClearSearchValue) {
+            setSearchValue('')
+            setSearchData([])
+        }
+    }, [onSelect])
 
     const searchMoreSuggestions = useCallback(
         async (value, skip) => {
             if (isAllDataLoaded) return
 
-            setLoading(true)
-            const data = await search(
-                client,
-                value,
-                null,
-                10,
-                skip
-            )
-            setLoading(false)
+            setMoreLoading(true)
+            const data = await search(client, value, null, 10, skip)
 
             if (data.length > 0) {
-                setData(prevData => [...prevData, ...data])
+                if (searchValue) {
```

Comment:
This way it would be a bit readable and skip code duplication

```
const uniqData = prevData => uniqBy([...prevData, ...data], 'value')
const dataHandler = searchValue ? setSearchData : setAllData

dataHandler(uniqData)
```